### PR TITLE
Issue2023curlybracket

### PR DESCRIPTION
**Pull request by AxelTWC**
Pull Request for #2023 
## Issue2023curlybracket

This code currently raises a warning and detects the following Extra Brackets:

- OPEN_ROUND_BRACKETS
- CLOSE_ROUND_BRACKETS
- OPEN_CURLY_BRACKETS
- CLOSE_CURLY_BRACKETS 

and outputs the warning number 1016. 

In order to remove the warning, the user would need to look at the warning message showing the line number of the extra bracket.

## Note
- For Developers, Quick Build the Compiler twice as there is an edit in the Umple Internal Parser Code.
- Multiple Test Cases has been added to test and solve this issue. 
- An extra user manual guideline has been added to show how to reproduce this warning and how to solve the warning.

### CODE REVIEW

**Comment by AxelTWC** [comment]
## Reasons for all the pushes above:
This Comment Explains for all the absurd amount of pushes above.

## Test Cases accidentally all have Class instead of class.
An example would be:
- Class x{}
It was changed to:
- class x{}

## Jenkins Giving Failed Warnings.
- It does not pass Jenkins as Jenkins only build the compiler once
- Since there is a change in the internal parser code, the compiler should be compiled twice in order for brackets to be detected.

---------------------------------------------------

Everything else should work properly.

File: cruise.umple/src/class/umple_classes.grammar [hunk 1]
```diff
@@ -29,12 +29,16 @@ associationClassDefinition : associationClass [name] { [[associationClassContent
...
 associationClassContent- :  [[comment]] | [[reqImplementation]] | [[classDefinition]] | [[position]] | [[displayColor]] | [[invariant]] | [[softwarePattern]] | [[depend]] | [[association]] | [[inlineAssociation]] | [[singleAssociationEnd]] | [[attribute]] | [[stateMachine]] | [[enumerationDefinition]] | ; | [[extraCode]]
 innerClass : [[innerStaticClass]] | [[innerNonStaticClass]]
 innerStaticClass : static [[classDefinition]]
 innerNonStaticClass : inner [[classDefinition]]
 
+openCurlBracket- : {
```
Comment (TimLethbridge) [inline cruise.umple/src/class/umple_classes.grammar]: Please remove these two unneeded rules

File: cruise.umple/src/class/UmpleInternalParser_CodeClass.ump [hunk 2]
```diff
@@ -64,7 +64,9 @@ class UmpleInternalParser
       shouldProcessAgain = shouldProcessAgain || (analysisStep == 1);
       return;
     }
-
+    //Reveal the current class token
+    //System.out.print("Token:" + t);
```
Comment (TimLethbridge) [inline cruise.umple/src/class/UmpleInternalParser_CodeClass.ump]: Please take out this debug code

File: cruise.umple/src/class/UmpleInternalParser_CodeClass.ump [hunk 3]
```diff
@@ -352,7 +357,8 @@ class UmpleInternalParser
     {
       return;
     }
-
+    //Reveals the current detected class token
```
Comment (TimLethbridge) [inline cruise.umple/src/class/UmpleInternalParser_CodeClass.ump]: Please take out this debug code

File: cruise.umple/src/class/UmpleInternalParser_CodeClass.ump [hunk 4]
```diff
@@ -523,6 +529,9 @@ class UmpleInternalParser
     {
       analyzeException(token,aClass);
     }
+    else if (token.is("checkForUnintendedBracket")){
```
Comment (TimLethbridge) [inline cruise.umple/src/class/UmpleInternalParser_CodeClass.ump]: This check should only be at the top level ... and hence should only be there once ...

File: cruise.umple/src/class/umple_classes.grammar [hunk 5]
```diff
@@ -16,7 +16,7 @@ externalDefinition : external [=interface]? [name] { [[classContent]]* }
 // An Interface can only contain method. See [*interfaceDefinition*]
 interfaceDefinition : interface [name] { [[depend]]* [[interfaceBody]] }
 
-// Associations can be declared outside the body of classes. 
+// Associations can be declared outside the body of classes.
```
Comment (TimLethbridge) [inline cruise.umple/src/class/umple_classes.grammar:19]: Please reset so there is no change to the space here

File: cruise.umple/src/class/umple_classes.grammar [hunk 6]
```diff
@@ -29,12 +29,16 @@ associationClassDefinition : associationClass [name] { [[associationClassContent
...
+classContent- :  [[comment]] | [[reqImplementation]] | [[innerClass]] | [[mixsetDefinition]] | [[distributable]] | [[proxyPattern]] | [[strictness]] | [[classDefinition]] | [[trace]] | [[emitMethod]] | [[templateAttributeDefinition]] | [[primitiveDefinition]] | [[portDefinition]] | [[portBindingDefinition]] | [[position]] | [[displayColor]] | [[abstract]] | [[keyDefinition]] | [[softwarePattern]] | [[depend]] | [[symmetricReflexiveAssociation]] | [[attribute]] | [[testCase]] | [[genericTestCase]] | [[testSequence]] | [[testClassInit]] | [[stateMachine]] | [[activeMethodDefinition]] | [[inlineAssociation]] | [[concreteMethodDeclaration]] | [[constantDeclaration]] | [[modelConstraint]] | [[invariant]] | ; | [[enumerationDefinition]] | [[exception]] | [[extraCode]]
```
Comment (TimLethbridge) [inline cruise.umple/src/class/umple_classes.grammar]: Please reset so there is no change to the line ... it is just a space

File: cruise.umple/src/en.error [hunk 7]
```diff
@@ -207,7 +207,8 @@
 1014: 3, "http://manual.umple.org/?W1014ExcludedMethodNotFoundInjection.html", Excluded method '{0}' cannot be found. The exclusion was ignored. ;
 # Issue 1483
 1015: 3, "http://manual.umple.org/?BasicStateMachines.html", Methods in standalone state machines might not be correctly generated. State machines can be used in traits or put directly in a class to achieve the same effect. ;
-
+1016: 3, "http://manual.umple.org/?PageBeingDeveloped.html", Extra Open Or Closing Bracket encountered. Processing terminated. All code beyond this point is ignored. ;
+1020: 3, "http://manual.umple.org/?PageBeingDeveloped.html", Extra closing curly bracket encountered. Processing terminated. All code beyond this point is ignored. ;
```
Comment (TimLethbridge) [inline cruise.umple/src/en.error]: Please take out 1020 and change 1016 so the 'Processing terminated and following is no longer there.

**Review by TimLethbridge — CHANGES_REQUESTED** [review]

**Review by TimLethbridge — APPROVED** [review]

**Pull request merged**